src/Service/VendorService/TheMovieDatabase/Message/ApiSearchMessage.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/**
+ * @file
+ * Contains ApiSearchMessage that contains data about a search from the data
+ * well.
+ */
+
+namespace App\Service\VendorService\TheMovieDatabase\Message;
+
+use App\Service\MoreInfoService\Types\IdentifierType;
+
+/**
+ * Class ApiSearchMessage.
+ */
+class ApiSearchMessage implements \JsonSerializable
+{
+    private $vendorId;
+    private $identifierType;
+    private $pid;
+    private $title;
+    private $year;
+    private $originalYear;
+    private $director;
+
+    /**
+     * ApiSearchMessage constructor.
+     *
+     * @param int|null    $vendorId
+     *   The vendor id
+     * @param string|null $identifierType
+     *   The identifier type
+     * @param string|null $pid
+     *   The pid identifier
+     * @param array       $meta
+     *   A meta array containing title, year, originalYear, director
+     */
+    public function __construct(int $vendorId = null, string $identifierType = null, string $pid = null, array $meta = [])
+    {
+        $this->vendorId = $vendorId;
+        $this->identifierType = $identifierType;
+        $this->pid = $pid;
+        $this->title = $meta['title'] ?? null;
+        $this->year = $meta['year'] ?? null;
+        $this->originalYear = $meta['originalYear'] ?? null;
+        $this->director = $meta['director'] ?? null;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Serialization function for the object.
+     */
+    public function jsonSerialize()
```

Comment:
Is this not the same as the default serializer (i might be missing something here)

Reply:
This exposes the private variables in the object to serialization